data-prepper-plugins/otel-proto-common/src/test/java/org/opensearch/dataprepper/plugins/otel/codec/OTelLogsProtoBufDecoderTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.otel.codec;
+
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.model.log.OpenTelemetryLog;
+import org.opensearch.dataprepper.model.event.Event;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class OTelLogsProtoBufDecoderTest {
+    private static final String TEST_REQUEST_LOGS_FILE = "test-otel-log.protobuf";
+    private static final String TEST_REQUEST_MULTI_LOGS_FILE = "test-otel-multi-log.protobuf";
+    private int count;
+    
+    public OTelLogsProtoBufDecoder createObjectUnderTest(boolean lengthPrefixedEncoding) {
+        return new OTelLogsProtoBufDecoder(lengthPrefixedEncoding);
+    }
+
+    private void validateLog(OpenTelemetryLog logRecord, final int severityNumber, final String time, final String spanId) {
+        assertThat(logRecord.getServiceName(), is("my.service"));
+        assertThat(logRecord.getTime(), is(time));
+        assertThat(logRecord.getObservedTime(), is(time));
+        assertThat(logRecord.getBody(), is("Example log record"));
+        assertThat(logRecord.getDroppedAttributesCount(), is(0));
+        assertThat(logRecord.getSchemaUrl(), is(""));
+        assertThat(logRecord.getSeverityNumber(), is(severityNumber));
+        assertThat(logRecord.getFlags(), is(0));
+        assertThat(logRecord.getSeverityText(), is("Information"));
+        assertThat(logRecord.getSpanId(), is(spanId));
+        assertThat(logRecord.getTraceId(), is("5b8efff798038103d269b633813fc60c"));
+        Map<String, Object> mergedAttributes = logRecord.getAttributes(); 
+        assertThat(mergedAttributes.keySet().size(), is(9)); 
+    }
+
+    @Test
+    public void testParse() throws Exception {
+        InputStream inputStream = OTelLogsProtoBufDecoderTest.class.getClassLoader().getResourceAsStream(TEST_REQUEST_LOGS_FILE);
```

Comment:
You could use protobuf builders to construct the binary data for the test cases. That way, we would see them in the test code to compare against the assertions.

Reply:
David suggested the same. I can add such test case as well. But I still see value in keeping this binary files based tests.